plugins/tokens/update.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const idSchema = joi.reach(schema.models.token.base, 'id');
+
+module.exports = () => ({
+    method: 'PATCH',
+    path: '/tokens/{id}',
+    config: {
+        description: 'Update a token',
+        notes: 'Update a specific tooken',
+        tags: ['api', 'secrets'],
+        auth: {
+            strategies: ['token', 'session'],
+            scope: ['user']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const tokenFactory = request.server.app.tokenFactory;
+            const userFactory = request.server.app.userFactory;
+            const username = request.auth.credentials.username;
+
+            return Promise.all([
+                tokenFactory.get(request.params.id),
+                userFactory.get({ username })
+            ])
+            .then(([token, user]) => {
+                if (!token) {
+                    throw boom.notFound('Token does not exist');
+                }
+
+                if (!user) {
+                    throw boom.notFound('User does not exist');
+                }
+
+                if (token.userId !== user.id) {
+                    throw boom.forbidden('User does not own token');
+                }
+
+                return user.tokens
+                    .then((tokens) => {
```

Comment:
Let's keep the indentation consistent. Some of the `.then` statements are indented and some aren't.